provider/rfc2136/rfc2136.go

```diff
@@ -439,3 +440,22 @@ func chunkBy(slice []*endpoint.Endpoint, chunkSize int) [][]*endpoint.Endpoint {
 
 	return chunks
 }
+
+func findMsgZone(ep *endpoint.Endpoint, m *dns.Msg, r *rfc2136Provider) {
+	sort.Slice(r.zoneName, func(i, j int) bool {
+		return len(strings.Split(r.zoneName[i], ".")) > len(strings.Split(r.zoneName[j], "."))
+	})
```

Comment:
This sort  will be called for each entry.

Reply:
fixed
https://github.com/kubernetes-sigs/external-dns/pull/3976/commits/3e59dc37197535a6f3b2b427435f8eba357a88f0